core/services/graph_service.py

```diff
@@ -703,17 +708,43 @@ async def extract_entities_from_text(
             content: Text content to process
             doc_id: Document ID
             chunk_number: Chunk number within the document
+            prompt_overrides: Optional EntityExtractionPromptOverride with customizations for prompts
+            override_is_image: Optional flag to override image detection based on metadata
 
         Returns:
             Tuple of (entities, relationships)
         """
         settings = get_settings()
 
-        # Limit text length to avoid token limits
-        content_limited = content[: min(len(content), 5000)]
-
-        # We'll use the Pydantic model directly when calling litellm
-        # No need to generate JSON schema separately
+        # Determine if this chunk is an image based solely on metadata flag
+        is_image = override_is_image if override_is_image is not None else False
+        # For images, send full base64 content; for text, truncate to limit
+        if is_image:
+            content_limited = content
+        else:
+            content_limited = content[: min(len(content), 5000)]
+
+        # Build system message, differentiating between text and image inputs
+        if is_image:
+            # For images, instruct comprehensive visual and text interpretation
+            system_content = (
+                "You are a multi-modal extraction assistant for images. The input is a base64-encoded PNG of a document page. "
+                "Perform OCR to extract the text, and visually interpret all layout elements: tables, diagrams, charts, form fields, headings, and graphical icons. "
+                "Identify entities from both text and visuals, infer relationships depicted (e.g., flows, hierarchies, links), and draw logical conclusions from the combined information. "
+                "For entities, include their label and type (e.g., PERSON, ORGANIZATION, LOCATION, CONCEPT). "
+                "For relationships, output a JSON list of objects with source, target, and relationship fields. "
+                "Respond only with valid JSON representing the extracted entities and relationships."
+            )
+        else:
+            # For text, use standard extraction instructions
+            system_content = (
+                "You are an entity extraction and relationship extraction assistant. "
```

Comment:
We should keep the original system prompt pls! I tested it and works well. Also maybe for images, we could just append saying context is image. That should do the job.